spec/requests/vendor_api/v1.1/post_withdraw_or_decline_application_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'rails_helper'
+
+RSpec.describe 'Vendor API - POST /applications/:application_id/withdraw', type: :request do
+  include VendorAPISpecHelpers
+  include CourseOptionHelpers
+
+  before { stub_const('VendorAPI::VERSION', '1.1') }
+
+  it_behaves_like 'an endpoint that requires metadata', '/withdraw', '1.1'
+
+  it 'withdraws an application' do
+    application_choice = create_application_choice_for_currently_authenticated_provider(
+      status: 'awaiting_provider_decision',
+    )
+
+    post_api_request "/api/v1.1/applications/#{application_choice.id}/withdraw"
+
+    expect(response).to have_http_status(:ok)
+    application = parsed_response['data']['attributes']
+    expect(application['status']).to eq 'withdrawn'
+    expect(application['withdrawn_or_declined_for_candidate_by_provider']).to eq true
```

Comment:
could you also add an expectation for the timestamp being set at withdrawn/declined?